File: Java-Core/src/main/java/presto/java/JavaNativeCategoryBinding.java
package presto.java;

import presto.grammar.NativeCategoryBinding;
import presto.utils.CodeWriter;

public class JavaNativeCategoryBinding extends NativeCategoryBinding {

	JavaIdentifierExpression expression;
	
	public JavaNativeCategoryBinding(JavaIdentifierExpression expression) {
		this.expression = expression;
	}

	public JavaIdentifierExpression getExpression() {
		return expression;
	}
	
	@Override
	public void toDialect(CodeWriter writer) {
		writer.append("Java: ");
		expression.toDialect(writer);
	}
}

File: Java-CodeServer/src/main/resources/js/prompto/java/JavaNativeCategoryBinding.js
var NativeCategoryBinding = require("./../grammar/NativeCategoryBinding").NativeCategoryBinding;

function JavaNativeCategoryBinding(expression) {
	NativeCategoryBinding.call(this);
	this.expression = expression;
	return this;
}

JavaNativeCategoryBinding.prototype = Object.create(NativeCategoryBinding.prototype);
JavaNativeCategoryBinding.prototype.creator = JavaNativeCategoryBinding;

JavaNativeCategoryBinding.prototype.toDialect = function(writer) {
    writer.append("Java: ");
    this.expression.toDialect(writer);
};

exports.JavaNativeCategoryBinding = JavaNativeCategoryBinding;